From e1dbfc63ed365f324feaf59ef8ffc0c7e9ba1ddd Mon Sep 17 00:00:00 2001
From: rigaz29 <ryanbagas27@gmail.com>
Date: Sun, 23 Aug 2026 23:55:19 +0000
Subject: [PATCH 2/2] Revert "linkerconfig: izinkan namespace vendor memuat
 dari /system (non-Treble)"

This reverts commit 8be2dfd485a24d35879e3a3b7544c2eed6f804b5.
---
 contents/namespace/vendordefault.cc | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/contents/namespace/vendordefault.cc b/contents/namespace/vendordefault.cc
index 27a5303..a866839 100644
--- a/contents/namespace/vendordefault.cc
+++ b/contents/namespace/vendordefault.cc
@@ -45,24 +45,9 @@ Namespace BuildVendorNamespace([[maybe_unused]] const Context& ctx,
   ns.AddSearchPath("/vendor/${LIB}/hw");
   ns.AddSearchPath("/vendor/${LIB}/egl");
 
-  // Perangkat non-Treble: /vendor cuma direktori di dalam /system, HAL-nya
-  // passthrough, dan tidak ada VNDK yang memisahkan keduanya. Blob vendor di
-  // sini memang menaut pustaka platform, dan shim yang menjembataninya harus
-  // ikut tinggal di /system karena dependensinya (libsensor, libandroid,
-  // libstagefright, libmedia) tidak punya salinan vendor.
-  //
-  // Tanpa dua baris di bawah, mm-qcamera-daemon mati sebelum sempat jalan:
-  //   library "/system/lib/libshim_camera.so" ... is not accessible for the
-  //   namespace: [name="(default)", permitted_paths="/odm:/vendor:/system/vendor"]
-  // dan karena daemonnya tidak ada, get_number_of_cameras() mengembalikan -1
-  // sehingga CameraModule::init() memanggil setCapacity(0xFFFFFFFF) dan
-  // cameraserver abort berulang.
-  ns.AddSearchPath("/system/${LIB}");
-
   ns.AddPermittedPath("/odm");
   ns.AddPermittedPath("/vendor");
   ns.AddPermittedPath("/system/vendor");
-  ns.AddPermittedPath("/system");
 
   ns.GetLink("rs").AddSharedLib("libRS_internal.so");
   ns.AddRequires(base::Split(Var("LLNDK_LIBRARIES_VENDOR", ""), ":"));
-- 
2.43.0

